🎨feat : Socket module 정리 및 manager control 추가

raymondanythings opened the pull request:
## ✅ PR 체크리스트
-  코드 스타일 가이드 규칙을 준수했어요.

## PR 타입
설명하실 변경 사항은 어떤 종류인가요?

<!-- 해당하는 한가지 타입에 체크해 주세요. -->
- [ ] 버그 수정
- [x] 기능 추가
- [x] 코드 스타일 업데이트
- [ ] 기타

<!-- 빈칸에 이슈번호를 입력해 주세요. 이슈번호를 모를 경우 추후 추가 가능하며, 그냥 넘어가셔도 좋습니다. -->
## 이슈 번호 : #3


## 해당 PR이 프로젝트에 큰 영향이 있나요?
<!-- 해당하는 한가지 타입에 체크해 주세요. -->
- [x] Yes
- [ ] No

## 작업 내용
- socket manager를 활용하여 namespace 분기처리

### 추가 설명

유저 데이터를 저장하기 위해 redis scheme 생각해야함.

raymondanythings committed:
♻️refactor : Io -> Manager로 재 변경

- Socket 신규 매소드 타입 추론 안되던 문제 해결
- Manager를 클래스화 하여 맴버변수로 기 연결된 소켓 상태 관리
- socket 이벤트 리스너 생성 시 기존 리스너가 존재할 경우 체크

raymondanythings committed:
Merge branch 'master' into feature/#3-socket-optimization

raymondanythings committed:
🎨feat : socket 바인더 -> hooks로 변경

raymondanythings committed:
🎨feat : Socket 대기실, 방입장 namespace 분기

cheonkyu0 commented on src/screen/Home.tsx:
```diff
@@ -11,17 +11,16 @@ interface NicknameInput {
   const { register: roomRegister, handleSubmit: roomSubmit } = useForm<JoinRoomInput>()
   const { register: nicknameRegister, handleSubmit: nicknameSubmit } = useForm<NicknameInput>({
     defaultValues: {
       nickname: localStorage.getItem('plug_nickname') || '',
```
다음 리뷰 때
 localStorage, sessionStorage, cookie 차이에 대해 언급해주면 좋을거 같네요

raymondanythings committed:
🎨feat : 로그인페이지 apollo 셋업

raymondanythings committed:
🎨feat : 로그인 관련 로직 추가

- useSocket Error 핸들러 추가
- login, home screen -> gql query 추가
- user store 활용, 유저객체 핸들링

raymondanythings committed:
🚧chore : node wrtc 초기세팅

cheonkyu0 commented on src/layout/SocketLayout.tsx:
```diff
@@ -1,27 +1,52 @@
+import store from 'store/index'
+import { GetMeQuery } from '__api__/types'
+
+const GET_ME = gql`
```
graphQl 를 도입 하셨네요.
